### PR TITLE
Save race data to columns and properties.

### PR DESCRIPTION
**Pull request by ashercodeorg**
This PR causes the race data to be stored redundantly in the properties `races` key and the `urm` and `races` columns. Upcoming PRs will backfill the `urm` and `races` columns, then stop populating the properties `races` key, then eliminate the data in the properties `races` key, then remove the properties `races` key as a SerializedAttribute.

LOCAL TESTING:
- [x] The race interstitial shows and doesn't show when appropriate.
- [x] The responses to the race interstitial are persisted to the properties `races` key.
- [x] The responses to the race interstitial are persisted to the `urm` and `races` columns.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
FYI @bakerfranke and @bencodeorg.

**Comment by ashercodeorg** [comment]
Adding @hamms and @jeremydstone since @ewjordan is out.

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -374,9 +374,24 @@ def hash_email
...
+    races_array = races_comma_separated.split(',')
+    return nil if races_array.empty?
+    return nil if (races_array & ['opt_out', 'nonsense', 'closed_dialog']).any?
+    return true if (races_array & ['black', 'hispanic', 'hawaiian', 'american_indian']).any?
+    false
+  end
+
+  def sanitize_and_set_race_data
     return unless property_changed?('races')
+    return unless races
```
Comment (Hamms) [inline dashboard/app/models/user.rb]: won't this mean that users with a `races` property but not a `races` column will never get the column populated?

File: dashboard/app/models/user.rb [hunk 2]
```diff
@@ -544,6 +564,13 @@ def update_with_password(params, *options)
     end
   end
 
+  def update_without_password(params, *options)
+    if params[:races]
+      update_column(:races, params[:races].join(','))
+    end
+    super
+  end
```
Comment (Hamms) [inline dashboard/app/models/user.rb]: why is this necessary? Does this method skip the `before_save` handlers?

File: dashboard/app/models/user.rb (hunk 1 above)
Comment (ashercodeorg) [reply dashboard/app/models/user.rb]: Note that, because of overloaded methods, both `property_changed('races')` and `races` appear to refer to the SerializedAttribute key/value pair, not the column. That said, let me fix this to be less hacky.

File: dashboard/app/models/user.rb (hunk 2 above)
Comment (ashercodeorg) [reply dashboard/app/models/user.rb]: Not sure why it ended up there. Removed.

**Comment by ashercodeorg** [comment]
PTAL.

File: dashboard/app/models/user.rb [hunk 3]
```diff
@@ -374,9 +374,30 @@ def hash_email
...
+
+  def sanitize_and_set_race_data
     return unless property_changed?('races')
+    # Though not possible from the UI, it seems wise to support manual clearing of this field. Thus
+    # we check for this edge case.
+    if races.nil?
+      update_column(:races, nil)
+      update_column(:urm, nil)
+      return
+    end
```
Comment (Hamms) [inline dashboard/app/models/user.rb:398]: :+1:

File: dashboard/app/models/user.rb (hunk 2 above)
Comment (ashercodeorg) [reply dashboard/app/models/user.rb]: And now I remember... Though the before_save handlers make this redundant as-is, the column is not set independently of the properties key, making it impossible to stop setting the properties key (until something like this is done). Doing so in another PR.
